### ISSUE TITLE
Split project into multiple packages

### ISSUE THREAD

**Issue by rvock** [mysql-workbench-schema-exporter/mysql-workbench-schema-exporter#153]
This project currently has the code for all exporters/formatters available:
- Doctrine1 YAML
- Doctrine2 YAML
- Doctrine2 Annotations
- Doctrine2 ZF2
- ZendDB
- RendRest
- Sencha ExtJS3
- Sencha ExtJS4
- Propel XML
- Propel YAML
- Node Sequelize

Some of these projects might be outdated. I only use the Doctrine 2 Annotations exporter.

I suggest to split this project up into multiple packages with one base-package. This way it would be possible to only require one exporter using composer:
```
composer require mysql-workbench-schema-exporter/doctrine-2-annotations
```

This package would depend upon `mysql-workbench-schema-exporter/core` which contains the base models.

We could still add one package `mysql-workbench-schema-exporter/mysql-workbench-schema-exporter` which includes all exporters.

Splitting the projects would have multiple benefits:
- It is possible to release new versions to packagist only for some exporters
- Breaking changes to the core won't affect old/legacy exporters
- Smaller code-base in each project

Who is still working on this project and who has push-permissions? We could set up a new github organisation mysql-workbench-schema-exporter and add the current contributors.

It's also possible to transfer this repository to the organisation:
https://help.github.com/articles/transferring-a-repository/

**Comment by johmue** [comment]
Hi Robert,

This sounds like a very good idea to me since I have neither contributed nor used the library for ages. @tohenk what's your oppinion on this? Shall we rename the project as well (you'll lose maybe some blog posts with documentation but you'll be able to start from scratch)?

**Comment by tohenk** [comment]
I have no objection. We could migrate this project as organization.

**Comment by rvock** [comment]
I have created a new organisation [mysql-workbench-schema-exporter](https://github.com/mysql-workbench-schema-exporter) and invited you two as owners.

@johmue could you transfer this repository as described in [Transferring a repository](https://help.github.com/articles/transferring-a-repository/#transferring-from-a-user-to-an-organization)? The complete commit history and all issues/wiki entries etc will be transferred.

You can transfer the repository in the [repository settings](https://github.com/johmue/mysql-workbench-schema-exporter/settings#transfer_button)

After transferring, I could set up the new repositories with the dedicated projects and keep the commit history for each project.

**Comment by johmue** [comment]
Done.

**Comment by Spy-Seth** [comment]
This is a good a idea. The old Zend adapter should not block the development of project. :+1:

**Comment by rvock** [comment]
Thanks for transferring the repository. I have split up the project into multiple repositories. The git history for the new repositories is rewritten, so it's still possible to see the diffs:
https://github.com/mysql-workbench-schema-exporter/doctrine2-exporter/commits/master

I wasn't able to add them to packagist, yet, because the vendor key is owned by @Easen But I have contacted him and they should be there in a few days.

For now, it's possible to set up a new project using Composer and only depend on one of the exporters (which will in turn require the base project):
```json
{
	"name": "rvock/mwb.test",
	"description": "Test mysql-workbench-schema-exporter",
	"type": "project",
	"license": "MIT",
	"minimum-stability": "stable",
	"repositories": [{
		"type": "vcs",
		"url": "https://github.com/mysql-workbench-schema-exporter/mysql-workbench-schema-exporter.git"
	}, {
		"type": "vcs",
		"url": "https://github.com/mysql-workbench-schema-exporter/doctrine2-exporter.git"
	}],
	"require": {
		"mysql-workbench-schema-exporter/doctrine2-exporter": "~3.0.0"
	}
}

```

Once everything is on packagist, it's not required to add the repository-settings to composer.json

**Comment by Easen** [comment]
@rvock I have added you a maintainer to the packagist and I've updated the repo URL too.

**Comment by rvock** [comment]
I have set up the other packages on packagist and now they can be installed via composer. I've also configured the github-hooks so that pushing a new tag will automatically update the versions at packagist.

Example:
https://packagist.org/packages/mysql-workbench-schema-exporter/doctrine2-exporter

Next step should be the documentation. I will create a README.md for each subproject and move the project-specific options to that readme.
